### ISSUE TITLE
Concurrent downloads send progress updates to all progress callbacks

### ISSUE THREAD

**Issue by respectTheCode** [itinance/react-native-fs#811]
When 2 concurrent downloads are running the progress callback for each download will be called for progress from either download.

The if statement in the progress callback bellow should never be true.

```
const fileDownload = fs.downloadFile({
	fromUrl,
	toFile,
	progress: progress => {
		if (progress.jobId != fileDownload.jobId) {
			console.log(
				"jobId mismatch",
				progress.jobId,
				fileDownload.jobId
			);
			return;
		}
```

This happens because the concurrent downloads all broadcast the same `DownloadProgress` event across the bridge.

**Comment by creambyemute** [comment]
Maybe related:

If I have multiple concurrent file downloads (for signature picutres, pictures and pdfs) with the following code:

```javascript
const result = RNFS.downloadFile({
    fromUrl: `${Constants.DatafilesApi.GET_FILE_ONLY_ENDPOINT}${file.id}`,
    toFile: `${signaturesPath}${file.id}.${file.extension}`,
    headers: {
        Authorization: token
    },
    discretionary: true,
    cacheable: true
});
const response = await result.promise;
```

It sometimes writes the content to the wrong file/destination (for example the signature is written to a picturePath instead of desired signaturePath.

When swapping RNFS.downloadFile with RNFetchBlob on the other hand, it works as expected, the images/signature are always saved to the correct path with the correct name.

**Comment by cinjon** [comment]
Confirmed that this is an issue.

**Comment by itinance** [comment]
Thanks to #842, I have published 2.16.4 with the fix for this.
I am going to close this issue, but pls feel free to reopen if the issue persists

**Comment by MaxInMoon** [comment]
Same issue on 2.16.6 for me on Android, I am the only one?
